packages/@react-aria/utils/src/runAfterTransition.ts

```diff
@@ -71,7 +71,11 @@ function setupGlobalEvents() {
   document.body.addEventListener('transitionend', onTransitionEnd);
 }
 
-setupGlobalEvents();
+if (document?.readyState !== 'loading') {
```

Comment:
I'm not sure what version of TypeScript this project is using, but I opted for optional chaining for the first approach.